mock/app.go

```diff
@@ -0,0 +1,120 @@
+package mock
+
+import (
+	"encoding/json"
+	"fmt"
+
+	abci "github.com/tendermint/abci/types"
+	dbm "github.com/tendermint/tmlibs/db"
+	"github.com/tendermint/tmlibs/log"
+
+	bam "github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// NewApp creates a simple mock kvstore app for testing.
+// It should work similar to a real app.
+// Make sure rootDir is empty before running the test,
+// in order to guarantee consistent results
+func NewApp(logger log.Logger, rootDir string) (abci.Application, error) {
```

Comment:
lets consolidate with examples/kvstore.

Maybe best to put this as `baseapp/test_app.go` ? examples/kvstore can borrow from there ?

Reply:
This was from examples/kvstore.
That code was a little broken, and main is no good for tests.
kvstore can use the mock app